src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2825,6 +2838,133 @@ private void ReinferMethodAndVisitArguments(BoundCall node, TypeWithState receiv
             SetResult(node, GetReturnTypeWithState(method), method.ReturnTypeWithAnnotations);
         }
 
+        private bool IsStaticEqualsMethod(MethodSymbol method)
+        {
+            if (method.Equals(compilation.GetSpecialTypeMember(SpecialMember.System_Object__EqualsObjectObject)))
+            {
+                return true;
+            }
+            else if (method.Equals(compilation.GetSpecialTypeMember(SpecialMember.System_Object__ReferenceEquals)))
```

Comment:
>if (method.Equals(compilation.GetSpecialTypeMember(SpecialMember.System_Object__ReferenceEquals))) [](start = 17, length = 98)

nit: `else if` can be simplified to `else { return method.Equals(compilation.GetSpecialTypeMember(SpecialMember.System_Object__ReferenceEquals)); } `